docs/src/guides/languages/python.md

```diff
@@ -9,4 +9,89 @@ tags:
 # :simple-python: Python
 <!-- markdownlint-enable single-h1 -->
 
-[TODO](https://github.com/input-output-hk/catalyst-ci/issues/79)
+## Introduction
+
+<!-- markdownlint-disable max-one-sentence-per-line -->
+!!! Tip
+    If you're just looking for a complete example,
+    [click here](https://github.com/input-output-hk/catalyst-ci/blob/master/examples/python/Earthfile).
+    This guide will provide detailed instructions for how the example was built.
+<!-- markdownlint-enable max-one-sentence-per-line -->
+
+This guide will get you started with using the Catalyst CI to work with Python based projects.
+
+To begin, clone the Catalyst CI repository:
+
+```bash
+git clone https://github.com/input-output-hk/catalyst-ci.git
+```
+
+Navigate to `examples/python` to find a basic Python project, with the `Earthfile` in it.
+This is the `Earthfile` we will be building in this guide.
+You can choose to either delete the file and start from scratch,
+or read the guide and follow along in the file.
+
+## Building the Earthfile
+
+<!-- markdownlint-disable max-one-sentence-per-line -->
+!!! Note
+    The below sections will walk through building our `Earthfile` step-by-step.
+    In each section, only the fragments of the `Earthfile` relative to that section are displayed.
+    This means that, as you go through each section, you should be cumulatively building the `Earthfile`.
+    If you get stuck at any point, you can always take a look at the
+    [example](https://github.com/input-output-hk/catalyst-ci/blob/master/examples/python/Earthfile).
+<!-- markdownlint-enable max-one-sentence-per-line -->
+
+### Prepare base builder
+
+```Earthfile
+VERSION 0.7
+
+builder:
+    FROM ./../../earthly/python+python-base
+
+    COPY --dir ./src .
+    DO ./../../earthly/python+BUILDER
+```
+
+The first target `builder` is responsible to prepare an already configured Python environment,
+instal all needed tools and dependencies.
+Every Python project must be a [poetry](https://python-poetry.org) based project,
+so it is mandatory to have `pyproject.toml` file in the root dir of the project.
+
```

Comment:
Also mandatory to vendor the dependencies. `poetry.lock` needs to exist and be up-to-date.

Reply:
fixed